data/summary/types/cgm.go

```diff
@@ -231,6 +246,26 @@ func (s *CGMStats) Update(ctx context.Context, cursor fetcher.DeviceDataCursor)
 	return nil
 }
 
+func (B *CGMBucketData) CalculateVariance(value float64, duration float64) float64 {
+	var mean float64 = 0
+	if B.TotalMinutes > 0 {
+		mean = B.TotalGlucose / float64(B.TotalMinutes)
+	}
+
+	weight := float64(B.TotalMinutes) + duration
+	newMean := mean + (duration/weight)*(value-mean)
+	return B.TotalVariance + duration*(value-mean)*(value-newMean)
+}
+
+func (B *CGMBucketData) CombineVariance(newBucket *CGMBucketData) float64 {
```

Comment:
A note that this uses https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Parallel_algorithm

would be helpful for future reviewers, when verifying.

Reply:
Wikipedia can change at any time, archive.org link? copy of the text block inside the comment? How would you preserve this in the comment?

Reply:
See previous response

Reply:
Linked to the relevant wikipedia section in a comment.